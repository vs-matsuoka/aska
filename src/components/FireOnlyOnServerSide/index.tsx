import React, { useEffect, useState } from 'react';

const FireOnlyOnServerSide = ({ children }: { children: React.ReactNode }) => {
  const [isServerSide, setIsServerSide] = useState(true);
  useEffect(() => {
    setIsServerSide(typeof window === 'undefined');
  }, []);

  return <>{isServerSide ? children : <></>}</>;
};

export default FireOnlyOnServerSide;
