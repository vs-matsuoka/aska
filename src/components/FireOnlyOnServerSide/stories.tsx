import { Story, Meta } from '@storybook/react';
import FireOnlyOnServerSide from '.';

export default {
  title: 'FireOnlyOnServerSide',
  component: FireOnlyOnServerSide
} as Meta;

export const Default: Story = () => <FireOnlyOnServerSide>YO</FireOnlyOnServerSide>;
