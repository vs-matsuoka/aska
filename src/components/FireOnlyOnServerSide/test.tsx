import { expect, describe, it } from '@jest/globals';
import { render } from '@testing-library/react';
import { renderToString } from 'react-dom/server';

import FireOnlyOnServerSide from '.';

describe('<FireOnlyOnServerSide />', () => {
  it('should not render on ServerSide', () => {
    const ui = <FireOnlyOnServerSide>YO</FireOnlyOnServerSide>;
    const container = document.createElement('div');
    container.innerHTML = renderToString(ui);
    document.body.appendChild(container);

    expect(container.innerHTML).toMatch('YO');

    expect(container.firstChild).toMatchSnapshot();
  });
  it('should render on ClientSide', () => {
    const { container } = render(<FireOnlyOnServerSide>YO</FireOnlyOnServerSide>);

    expect(container.innerHTML).toMatch('');

    expect(container.firstChild).toMatchSnapshot();
  });
});
