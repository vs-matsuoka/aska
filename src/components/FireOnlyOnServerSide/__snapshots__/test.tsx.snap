// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FireOnlyOnServerSide /> should not render on ServerSide 1`] = `YO`;

exports[`<FireOnlyOnServerSide /> should render on ClientSide 1`] = `null`;
